packages/xod-client/src/core/crashReporterMiddleware.js

```diff
@@ -0,0 +1,23 @@
+import { addError } from '../messages/actions';
+import composeMessage from '../messages/composeMessage';
+
+export default () => next => action => {
+  try {
+    return next(action);
+  } catch (err) {
+    // eslint-disable-next-line no-console
+    console.error(err);
+
+    return next(
+      addError(
+        composeMessage(
+          'Error',
+          err.message,
```

Comment:
Oops. The `composeMessage` is legacy. Better directly construct the stanza. It’s more readable:

```js
{
  title: 'Error',
  note: err.message,
  solution: 'The error was not caught, which is a bug. Report the issue to XOD developers.'
}
```